tests/manage/z_cluster/nodes/test_disk_failures.py

```diff
@@ -314,11 +318,16 @@ def test_recovery_from_volume_deletion(
         osd_pvc.ocp.wait_for_delete(resource_name=osd_pvc_name)
 
         # Delete the OSD deployment
-        logger.info(f"Deleting OSD deployment {osd_deployment_name}")
-        osd_deployment.delete()
-        osd_deployment.ocp.wait_for_delete(
-            resource_name=osd_deployment_name, timeout=120
-        )
+        logger.info(f"Verifying deletion of OSD deployment {osd_deployment_name}")
+        try:
+            osd_deployment.ocp.wait_for_delete(
+                resource_name=osd_deployment_name, timeout=30
+            )
+        except TimeoutError:
+            osd_deployment.delete()
+            osd_deployment.ocp.wait_for_delete(
+                resource_name=osd_deployment_name, timeout=120
+            )
```

Comment:
Do we still need to delete deployment of osd in ocs 4.6

Reply:
As far as I know, we don't. That's why I changed it to the code above - I am just verifying that the OSD deployment has been deleted, and if not, it will delete it. Do you think we can skip it and even don't check that the OSD deployment has been deleted?

Reply:
I think with the updated ocs-osd-removal job, it will take care of deleting the deployment, prepare pods and PVC. If that's the case, we can remove the manual deletion steps for it, but keep the checks to verify if these resources have been deleted.

Reply:
I asked Servesha, and she confirmed that it's the case. The ocs-osd-removal job, it will take care of deleting the deployment, prepare pods and PVC. We just need to verify that these resources have been deleted.

Reply:
it looks ok with the current implementation - for clusters that are lower than 4.6, we explicitly delete the OSD deployment, pod and PVC and for 4.6 and above, we just validate their deletion